### PR TITLE
Don't send geofence notifications for users who don't have access to …

### PR DESCRIPTION
**Pull request by jcardus**
…the geofence

### CODE REVIEW

File: src/main/java/org/traccar/database/NotificationManager.java [hunk 1]
```diff
@@ -134,6 +134,12 @@ private void updateEvent(Event event, Position position) {
                         LOGGER.info("User {} notification blocked", user.getId());
                         return;
                     }
+                    if (event.getGeofenceId() != 0
+                        && cacheManager.getUserObjects(user.getId(), Geofence.class).stream()
+                            .noneMatch(g -> g.getId() == event.getGeofenceId())) {
```
Comment (tananaev) [inline src/main/java/org/traccar/database/NotificationManager.java:139]:
```suggestion
                                && cacheManager.getUserObjects(user.getId(), Geofence.class).stream()
                                .noneMatch(g -> g.getId() == event.getGeofenceId())) {
```

File: src/main/java/org/traccar/database/NotificationManager.java [hunk 2]
```diff
@@ -134,6 +134,12 @@ private void updateEvent(Event event, Position position) {
...
+                        LOGGER.info("Geofence {} not in user {}", event.getGeofenceId(), user.getId());
```
Comment (tananaev) [inline src/main/java/org/traccar/database/NotificationManager.java:140]: Probably better word something like "User X doesn't have access to geofence Y".

**Comment by tananaev** [comment]
If we're doing this, probably makes sense to do a check for maintenances as well.

**Comment by jcardus** [comment]
I copied the getUserObjects from getDeviceObjects, what does proxies and forward parameters do in getObjects?

**Comment by tananaev** [comment]
> I copied the getUserObjects from getDeviceObjects, what does proxies and forward parameters do in getObjects?

Have you tested all the cases? This question makes me worry that you haven't actually tested it properly.

**Comment by jcardus** [comment]
That's why I asked about it. Yes, this needs more testing. I'll close the PR for now.

**Comment by tananaev** [comment]
In the future please only send a pull request for review when it's ready and thoroughly tested. If you want to discuss something related to a change, you can open an issue instead.

**Pull request closed without merging**